Feature/IIWA-83

kirbur opened the pull request:
LicensesCard & SoftwareCard

kirbur committed:
IIWA-39 NavBar exists / is clickable

Commit:
IIWA-35: Add .gitignore for backend-api and Visual Studio.

kirbur committed:
IIWA-39 changed the style

kirbur committed:
IIWA-39 Reviewed comments and implemented changes

kirbur committed:
IIWA-39 removed unnecessary circle class

kirbur committed:
Merge pull request #1 from kirbur/feature/IIWA-39

IIWA-39 NavBar exists / is clickable

kirbur committed:
IIWA-41 reversed order of links for css layering

kirbur committed:
IIWA-41 changed width of tabs

kirbur committed:
IIWA-41 removed unecessary code

kirbur committed:
IIWA-41 fixed border radius on department tab

kirbur committed:
IIWA-41 removed unecessary lines

kirbur committed:
IIWA-41 Changed cursor on labels and ellipse color

kirbur committed:
IIWA-41 Repositioned navEllipse and navRectange

Commit:
added test comment

Commit:
created the PieChart component using recharts

kirbur committed:
IIWA-41 removed unecessary line

kirbur committed:
Merge pull request #2 from kirbur/feature/IIWA-41

Feature/iiwa 41

mtleonard120 committed:
Merge pull request #2 from kirbur/IIWA-21

IIWA-21

Commit:
made changes from code review

charkour committed:
IIWA-35: Run Scaffold-DbContext. Note: Make sure Window's User has db ownership.

charkour committed:
IIWA-35: Fixed namespace and registered ITInvetoryDBContext with dependency injection.

kirbur committed:
IIWA-31 Created Dropdown, needs style

charkour committed:
IIWA-35: Added EmployeeController template.

charkour committed:
IIWA-35: Run Scaffold DbContext after updated db modles.

charkour committed:
IIWA-35: Fix the namespace of models and create template EmployeeController.

Commit:
cost breakdown component

charkour committed:
IIWA-35: Re-add DI injection to Program. Added port 4001 for https in Startup. Added authorization to Controllers.

kirbur committed:
IIWA-21 Fixed navbar tab rectangle position

charkour committed:
IIWA-35: Add Automatic AD authentication when running IIS app. Add sample string API call but have to fill in proper username and domain for it to work properly.

charkour committed:
IIWA-35: Final (please!) creation of models to fit new db designs.

kirbur committed:
IIWA-21 fixed tab label alignment

charkour committed:
IIWA-35: Added OData for powerful querying from front-end calls.

mtleonard120 committed:
Merge pull request #3 from kirbur/IIWA-21

IIWA-21

charkour committed:
IIWA-35: Rebuild one model and the context after changing some columns.

Commit:
changed CostCard breakdown

Commit:
removed CostBreakdown.tsx

Commit:
dashboard table component

charkour committed:
Merge pull request #1 from charkour/feature/IIWA-35

Feature/iiwa 35

Commit:
changed some CSS

charkour committed:
IIWA-36: .gitignore fix. Search on Google: Untrack files all added to
git repository based on .gitignore.

charkour committed:
IIWA-36: Department cotroller. Comment code to make no infinite loop.

kirbur committed:
Merge branch 'develop' of https://github.com/cqlcorp/it-inventory into feature/IIWA-31

mtleonard120 committed:
Merge pull request #4 from charkour/feature/IIWA-20

Feature/iiwa 20

mcdowellalex committed:
changed couple things in CostCard.tsx

kirbur committed:
IIWA-31 styled dropdown

kirbur committed:
IIWA-31 can pass a function to dropdown

mcdowellalex committed:
fixed suggestions

kirbur committed:
IIWA-43 created service class

kirbur committed:
IIWA-31 made it more readable

kirbur committed:
Merge pull request #5 from kirbur/feature/IIWA-31

feature/IIWA-31

charkour committed:
IIWA-36: Added endpoint to fill in the department table on the dashboard.

charkour committed:
IIWA-36: Add ability to remove programs of cost 0. Renamed controller and routes to make more sense.

charkour committed:
Add ProgramOneTimeCost field because Dan mentioned he might want it.

Xiscoder committed:
IIWA-54:Built cost controller and build an endpoint within that controller to calculate costs from various tables

mcdowellalex committed:
-

Xiscoder committed:
IIWA-54 Added methods in models to allow for accesing of those fields in the databse.
Cleaned up my dashboard cost endpoint.
Dashboard endpoint now only returns programs and their plugins and now returns as an array

mcdowellalex committed:
added total, programs, and plugins

mcdowellalex committed:
added Card back

charkour committed:
IIWA-36: Make sure that program is not deleted when put into the total cost of the department.

mcdowellalex committed:
added headings and restyled

charkour committed:
IIWA-36: Add IsYearPerCost to return in DepartmentTableCost and remove FixedCost.

kirbur committed:
IIWA-43 Created an axios service

charkour committed:
Merge pull request #3 from charkour/feature/IIWA-36

Feature/IIWA-36

Xiscoder committed:
Merge pull request #1 from Xiscoder/feature/IIWA-54

Feature/iiwa 54

freakingawesome committed:
Add fluent migrator initial tables

freakingawesome committed:
Move connstring to envar

freakingawesome committed:
Add base connstring for migrations

freakingawesome committed:
Add .gitignore

freakingawesome committed:
Respect .gitignore

charkour committed:
IIWA-23: Add basic ProgramsController.

mcdowellalex committed:
detailing

mcdowellalex committed:
merge story branch into subtask branch

mcdowellalex committed:
added second pie chart and made custom legend

mcdowellalex committed:
added titles

mcdowellalex committed:
made changes Mike suggested

mcdowellalex committed:
minor change

charkour committed:
Update connection strings and add DateBought field to Program table via FluentMigration.

kirbur committed:
IIWA-43 Added token managment functions

mcdowellalex committed:
fixed some CSS

kirbur committed:
IIWA-43 Removed old login function

charkour committed:
Add script to generate test data and readme documentation.

kirbur committed:
IIWA-43 refreshToken re-trys the get/post request

charkour committed:
Revert "IIWA-23: Add basic ProgramsController."

This reverts commit f3b975a3210c8daace6e41973e152970122bca7a.

mtleonard120 committed:
Merge pull request #6 from charkour/feature/fluent-migrations

Feature/fluent migrations

mtleonard120 committed:
Merge pull request #5 from cqlcorp/feature/fluent-migrations

Add FluentMigrator to project

charkour committed:
IIWA-56: Add Program controller

Xiscoder committed:
IIWA-29: Creating endpoint for piechart(s) on dashboard. Added some fields to the models

mcdowellalex committed:
made pie charts more flexible

charkour committed:
Update field DateBought on Program table.

mcdowellalex committed:
-

charkour committed:
Fix migration ID. How do you learn if you don't make mistakes?

Xiscoder committed:
Pulling Kirsten's 22 branch

charkour committed:
IIWA-56: Software Table controller basics work. Need to order by activity and user settings now.

kirbur committed:
Merge branch 'develop' of https://github.com/cqlcorp/it-inventory into feature/IIWA-43

Xiscoder committed:
Working on programs controller

Xiscoder committed:
Merge pull request #2 from Xiscoder/feature/IIWA-29

IIWA-29: Creating endpoint for piechart(s) on dashboard. Added some f…

mcdowellalex committed:
testing it

mcdowellalex committed:
removed headings and restyled it

charkour committed:
Fix migration names to run in proper order.

Commit:
Merge remote-tracking branch 'Kirstin/feature/IIWA-22' into feature/IIWA-51

Xiscoder committed:
Added a class to be used to hold the return object for the license bar graph. Built endpoint for license controller

Commit:
IIWA-51 Update CostCard with title props

Commit:
IIWA-51 Add Icons in

mcdowellalex committed:
Merge pull request #3 from mcdowellalex/feature/IIWA-68

dashboard table component

mcdowellalex committed:
Merge pull request #5 from mtleonard120/feature/IIWA-51

Feature/iiwa 51

mcdowellalex committed:
detailed icons

kirbur committed:
IIWA-43 Can successfully get data

charkour committed:
IIWA-56: SoftwareTable controller now pins software from the UserSettings (in a hacky way), and limits the rest of the list to 10.

kirbur committed:
IIWA-43 commented out token stuff in get/post

kirbur committed:
Merge pull request #7 from kirbur/feature/IIWA-43

Feature/IIWA-43

mtleonard120 committed:
Merge pull request #12 from kirbur/feature/IIWA-19

Feature/iiwa 19

mcdowellalex committed:
-

charkour committed:
IIWA-56: Orders the SoftwareTable list by the most recent activity while also respecting the UserSettings. Left some TODOs to come back to later.

kirbur committed:
IIWA-22 Pulled from develop

kirbur committed:
IIWA-22 Merged from Alex/freature/IIWA-22

mcdowellalex committed:
added axios to cost card

mtleonard120 committed:
Merge pull request #11 from charkour/feature/fluent-migrations

Feature/fluent migrations

mtleonard120 committed:
Merge pull request #13 from cqlcorp/feature/fluent-migrations

Feature/fluent migrations

kirbur committed:
IIWA-83 Created BarChartCard to place on dashboard

charkour committed:
Merge remote-tracking branch 'CQL/develop' into feature/IIWA-23

charkour committed:
Merge pull request #3 from charkour/feature/IIWA-36

Feature/IIWA-36

Xiscoder committed:
Merged dev into IIWA-24

kirbur committed:
IIWA-83 Created SoftwareCard for dashboard

Xiscoder committed:
Merge IIWA-25 into dev

Xiscoder committed:
gitignorefix

mcdowellalex committed:
added axios code

kirbur committed:
IIWA-83 renamed BarChartCard to LicensesCard

Xiscoder committed:
Merged IIWA-45 into dev

mcdowellalex commented on src/frontend-admin/src/components/SoftwareCard/SoftwareCard.tsx:
```diff
@@ -0,0 +1,37 @@
+  useEffect(() => {
+    //TODO: find out endpoint name
+    axios.get("/Software", setTableData);
+  }, [setTableData]);
```
The dashboardTable.tsx should have the axios stuff in the actual component. To be able to use it on the department card with the dropdown, we could pass through the endpoint as a prop

mtleonard120 committed:
Merge pull request #10 from charkour/feature/IIWA-23

Feature/iiwa 23

kirbur committed:
IIWA-83 moved axios request into DashboardTable

kirbur committed:
IIWA-83 fixed licence endpoint

kirbur committed:
Created Dashboard page w/ flexbox

mtleonard120 committed:
Merge pull request #14 from kirbur/DashboardPage

Created Dashboard Page

Xiscoder committed:
Fixing data formating

kirbur committed:
IIWA-83 pulled from dev

mcdowellalex committed:
reverted back cause axios is going on the dashboard

mcdowellalex committed:
axios stuff going on the dashboard

Xiscoder committed:
Fixing merge conflicts

Xiscoder committed:
Fixing Merge Conflicts

Xiscoder committed:
uncomment migrator

kirbur committed:
IIWA-83 pulled from alex

kirbur committed:
IIWA-83 wired up pies

kirbur committed:
Merge branch 'feature/IIWA-23' of https://github.com/kirbur/it-inventory into feature/IIWA-83

mtleonard120 committed:
Merge pull request #8 from Xiscoder/feature/IIWA-25

Feature/iiwa 25

Xiscoder committed:
Merge remote-tracking branch 'CQL/develop' into feature/IIWA-24

Xiscoder committed:
Source control

mtleonard120 committed:
Merge pull request #7 from Xiscoder/feature/IIWA-24

Feature/iiwa 24

Xiscoder committed:
IIWA-60 untracked annoying files

Xiscoder committed:
Merge remote-tracking branch 'CQL/develop' into feature/IIWA-45

mtleonard120 committed:
Merge pull request #9 from Xiscoder/feature/IIWA-45

Feature/iiwa 45

charkour committed:
Merge pull request #4 from charkour/feature/IIWA-56

Feature/iiwa 56

charkour committed:
Merge remote-tracking branch 'CQL/develop' into feature/IIWA-22

kirbur committed:
IIWA-83 started wiring up dropdown

mcdowellalex committed:
added font folder
index.css has cql font on body

freakingawesome committed:
Add initial build

This will probably fail on a build error but that is legit so it is ok

freakingawesome committed:
Avoid tagging during build for now

kirbur committed:
IIWA-83 no more console errors

mcdowellalex committed:
moved content folder

freakingawesome committed:
Remove parfait straggler

Xiscoder committed:
stupid employee id error

Xiscoder committed:
Merge pull request #3 from Xiscoder/feature/IIWA-45

Added a class to be used to hold the return object for the license ba…

mtleonard120 committed:
Merge pull request #17 from Xiscoder/feature/IIWA-22

Feature/iiwa 22

mcdowellalex committed:
fix

freakingawesome committed:
Merge remote-tracking branch 'origin/develop' into feature/build-server-init

mtleonard120 committed:
Merge pull request #18 from mcdowellalex/fix-for-Chad

fix for chad

charkour committed:
Merge branch 'feature/IIWA-22' of https://github.com/charkour/it-inventory into feature/IIWA-22

charkour committed:
IIWA-56: Respect .gitignore

freakingawesome committed:
Merge remote-tracking branch 'origin/develop' into feature/build-server-init

freakingawesome committed:
Try fixing build

freakingawesome committed:
Configure automated dev deploy

freakingawesome committed:
Make deployment script executable

kirbur committed:
IIWA-83 correctly reads in list of depts

kirbur committed:
IIWA-83 visual studio changed files

Xiscoder committed:
Add gitignore

kirbur committed:
IIWA-83 pulled from dev

mcdowellalex committed:
fixed pie data and spacing

mcdowellalex committed:
fixed a couple things on the pie chart

mtleonard120 committed:
Merge pull request #20 from mcdowellalex/IIWA-28

Iiwa 28

mtleonard120 committed:
Merge pull request #15 from charkour/feature/IIWA-22

Feature/iiwa 22

mtleonard120 committed:
Merge pull request #16 from mcdowellalex/fonts

IIWA-81

kirbur committed:
IIWA-83 wired up pies

kirbur committed:
IIWA-83 pulled from dev

Xiscoder committed:
Login endpoints and some helper classes

Xiscoder committed:
Made requested changes

Xiscoder committed:
Merge pull request #4 from Xiscoder/feature/IIWA-60

Feature/iiwa 60

Xiscoder committed:
solved merge conflicts

mtleonard120 committed:
Merge pull request #21 from Xiscoder/feature/IIWA-26

Feature/iiwa 26

freakingawesome committed:
Merge remote-tracking branch 'origin/develop' into feature/build-server-init

freakingawesome committed:
Fix startup connection string

If you want to solve a git mystery, check out the history surrounding
this and PR #5, and try to figure out why the commits by Chad do not
show up in `git log src/backend-api/Startup.cs` unless you use the
`--follow` flag of `git log`.

I wonder if there was a bad merge somewhere and have spent a bit of time
trying to understand the problem, but I have to move on. Hopefully this
is the only bad thing that happened with whatever actually happened.

Good luck. We're all counting on you.

freakingawesome committed:
Overcome connection string envar config magic

freakingawesome committed:
Allow hosting wwwroot in iis

freakingawesome committed:
Serve index.html as fallback

This allows for the build server to layer the React front-end on top of
the wwwroot directory.

freakingawesome committed:
Notify when app deployment is underway

freakingawesome committed:
Add JWT Secret for dev site

freakingawesome committed:
Try fix dev site static files

freakingawesome committed:
Only run dev deploy on develop branch

freakingawesome committed:
Pffft. I don't want your build error emails

mtleonard120 committed:
Merge pull request #22 from cqlcorp/feature/build-server-init

Automatically deploy `develop` branch to dev site

charkour committed:
IIWA-83: Fix Costbreakdown Endpoint to return an array.

kirbur committed:
IIWA-83 wired up software table

kirbur committed:
IIWA-83 Pulled from dev

charkour committed:
Respect .gitignore again...

charkour committed:
IIWA-83: Update comment for endpoint documentation.

charkour committed:
Merge

kirbur committed:
Merge pull request #9 from charkour/feature/IIWA-83

Feature/iiwa 83

kirbur committed:
IIWA-83

kirbur committed:
IIWA-83 pulled Charlies changes

Xiscoder committed:
More data added

mtleonard120 committed:
Merge pull request #23 from Xiscoder/develop

More data added

kirbur committed:
IIWA-83 Fixed dashboard styling

kirbur committed:
IIWA-83 fixed dashboard styles

kirbur committed:
Merge branch 'develop' of https://github.com/cqlcorp/it-inventory into feature/IIWA-83

kirbur committed:
IIWA-83 fixed styles

mcdowellalex reviewed (approved):
looks good
